Update the host networking documentation for GA (#20630)
docker/docs: content/engine/network/drivers/host.md

This is because it doesn't require network address translation (NAT), and no "userland-proxy" is created for each port.

> The host networking driver works on Linux hosts, and is available with Docker Desktop version 4.34 and later.

Reviewer: suggestion:

```suggestion
The host networking driver is supported on Docker Engine (Linux only) and Docker Desktop version 4.34 and later.
```

## Docker Desktop

> Host networking is also supported on Docker Desktop version 4.34 and later for Mac,
> Windows, and Linux. To enable this feature, navigate to the **Resources**, **Network** tab
> in **Settings**, and then select **Enable host networking**.

Reviewer: ```suggestion
Host networking is supported on Docker Desktop version 4.34 and later.
To enable this feature:

1. Sign in to your Docker account in Docker Desktop.
2. Navigate to the **Settings** in Docker Desktop
3. Under the **Resources** tab, select **Network**.
4. Check the **Enable host networking** option.
```
